### PR TITLE
Cursor capture fixes

### PR DESCRIPTION
**Pull request by offalynne**
Prevents cursor capture (dropping if active) when the window is positioned outside of display bounds resulting in an unworkable state. Adds `blocked` member to the struct returned by `input_mouse_capture_get()` to indicate this state. Block is released and capture is recovered when the window reenters a workable state.

### CODE REVIEW

File: scripts/__input_system_tick/__input_system_tick.gml [hunk 1]
```diff
@@ -166,9 +159,19 @@ function __input_system_tick()
         ++_m;
     }
     
-    __input_release_multimonitor_cursor();
+    //Block mouse capture when the window state changes
+    if (_global.__mouse_capture && __input_window_changed())
```
Comment (JujuAdams) [inline scripts/__input_system_tick/__input_system_tick.gml]: Should this be `... && _global.__window_focus && __input_window_changed()`? I'm concerned about window state changing whilst the window is out of focus leading to `__input_window_changed()` returning `false` when the window is back in focus.

**Comment by offalynne** [comment]
Gotta ~~test re. review focus issue and~~ deworm Mac a bit so pls leave open for now

**Comment by offalynne** [comment]
OK we all good, thanks for code review @JujuAdams and for Mac smoke testing @tabularelf

**Comment by offalynne** [comment]
~~Still problematic on not-Windows for the time being~~

**Comment by offalynne** [comment]
All clear thanks @tabularelf

**Comment by JujuAdams** [comment]
Thanks both

**Pull request merged**